package.json

```diff
@@ -94,6 +102,30 @@
     "react-app-rewired": "^2.0.0",
     "react-scripts": "^2.0.3",
     "rimraf": "^2.6.2",
-    "source-map-explorer": "^1.6.0"
+    "source-map-explorer": "^1.6.0",
+    "ts-jest": "^23.10.3",
+    "ts-loader": "^5.2.1",
+    "typescript": "^3.1.1"
+  },
+  "jest": {
+    "collectCoverageFrom": [
+      "src/**/*.{ts,tsx}"
+    ],
+    "testMatch": [
+      "<rootDir>/src/**/__tests__/**/*.{ts,tsx}",
+      "<rootDir>/src/**/*.(spec|test).{ts,tsx}"
+    ],
+    "transform": {
+      "^.+\\.[tj]sx?$": "ts-jest"
+    },
+    "moduleFileExtensions": [
+      "ts",
+      "tsx"
+    ],
```

Comment:
These all only specify ts/tsx because they get merged with the already existing jest config that supports js